src/http/httpd.cc

```diff
@@ -423,7 +427,9 @@ future<> http_server::do_accepts(int which) {
 
 future<> http_server::do_accept_one(int which) {
     return _listeners[which].accept().then([this] (accept_result ar) mutable {
-        auto conn = std::make_unique<connection>(*this, std::move(ar.connection), std::move(ar.remote_address));
+        auto conn = std::make_unique<connection>(*this,
+            std::move(ar.remote_address), ar.connection.local_address(),
+                    std::move(ar.connection));
```

Comment:
This is problematic -- the order of `ar.connection.local_address()` and `std::move(ar.connection)` is undefined, so it may happen that the `.local_address()` is called on moved-from `ar.connection`